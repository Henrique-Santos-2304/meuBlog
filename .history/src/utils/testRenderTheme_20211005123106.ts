import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<ThemeProvider>{children}</ThemeProvider> );
};
