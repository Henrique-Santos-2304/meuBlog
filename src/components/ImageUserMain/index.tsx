/* eslint-disable @next/next/no-img-element */
import { Flex, useMediaQuery } from "@chakra-ui/react";
import { Img } from "@chakra-ui/react";
import IconsContact from "components/IconsContact";

const ImageUserMain = () => {
  const [isGreatherThan768] = useMediaQuery("(max-width: 768px)");
  return (
    <Flex
      align="center"
      justify="center"
      direction="column"
      mt={["1rem", null, "-5rem"]}
    >
      <Img
        boxShadow="0 0 20px red, 0 0 20px white"
        boxSize={["20rem", null, "22rem", "24rem"]}
        borderRadius="full"
        src="https://res.cloudinary.com/defnibbpl/image/upload/v1633360607/large_henrique_techs_89a8aae1bc.jpg"
      />
      {isGreatherThan768 && <IconsContact />}
    </Flex>
  );
};

export default ImageUserMain;
