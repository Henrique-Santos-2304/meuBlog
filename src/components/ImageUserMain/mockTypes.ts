export const mockTypes = {
  data: {
    alt: "photo",
    ariaLabel: "uma img qualquer",
    imageUser: {
      name: "photo",
      url: "img.imagen.png",
    },
  },
  links: {
    name: "Henrique dos Santos",
    profission: "Desenvolvedor FrontEnd",
    platform: "Web, Mobile, Desktop",
    linksSocials: [],
    email: {
      title: "icons",
      email: "henrique.multitech@gmail.com",
      ariaLabel: "Envio de email via site",
      id: "name",
    },
  },
};
