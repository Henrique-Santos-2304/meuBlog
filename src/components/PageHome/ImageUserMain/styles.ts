import styled from "styled-components";

export const PhotoContent = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin-top: -4rem;

  @media (min-width: ${({ theme }) => theme.media.medium}) {
    margin-top: -5rem;
  }
`;

export const ImageUser = styled.img`
  width: 16rem;
  height: 16rem;
  object-fit: cover;
  border-radius: 50%;
`;
