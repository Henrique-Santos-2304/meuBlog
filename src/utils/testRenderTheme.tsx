import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
