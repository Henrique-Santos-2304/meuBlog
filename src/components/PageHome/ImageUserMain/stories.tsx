import { Story, Meta } from "@storybook/react";
import ImageUserMain from ".";

export default {
  title: "ImageUserMain",
  component: ImageUserMain,
} as Meta;

export const Basic: Story = (args) => <ImageUserMain {...args} />;

Basic.args = {
  title: "title basic",
  description: "description basic",
};

export const Default: Story = (args) => <ImageUserMain {...args} />;
