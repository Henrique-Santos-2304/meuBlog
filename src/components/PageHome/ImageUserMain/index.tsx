/* eslint-disable @next/next/no-img-element */
import * as S from "./styles";

import {
  queryHome_home_userImage,
  queryHome_home_boxMain,
} from "graphql/typesFinal/queryHome";

import IconsContact from "components/PageHome/IconsContact";
import HandleMediaQuery from "components/globalComponents/MediaQuery";

type props = {
  data: queryHome_home_userImage;
  links: queryHome_home_boxMain;
};

const ImageUserMain = (props: props) => {
  const { data, links } = props;

  const iconLink = {
    linksSocials: links.linksSocials,
    email: links.email,
  };

  return (
    <S.PhotoContent aria-label="Foto de um Programador/ Desenvolvedor  ">
      <S.ImageUser
        width={100}
        height={100}
        src={data.imageUser.url}
        alt={data.alt}
        aria-label={data.ariaLabel}
      />
      <HandleMediaQuery responsive="max-width" asWidth="767px" show={true}>
        <IconsContact contacts={iconLink} />
      </HandleMediaQuery>
    </S.PhotoContent>
  );
};

export default ImageUserMain;
