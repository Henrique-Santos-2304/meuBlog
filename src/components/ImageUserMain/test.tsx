import ImageUserMain from ".";
import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";

jest.mock("@chakra-ui/react", () => {
  const originalModule = jest.requireActual("@chakra-ui/react");

  return {
    __esModule: true,
    ...originalModule,
    useMediaQuery: jest.fn().mockImplementation(() => [true]),
  };
});

describe("<ImageUserMain />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<ImageUserMain />);
    expect(container).toMatchSnapshot();
  });
  it("should have a image on the screen ", () => {
    renderTheme(<ImageUserMain />);
    expect(screen.getByRole("img")).toBeInTheDocument();
  });
  it("should have a image on the screen ", () => {
    renderTheme(<ImageUserMain />);
    screen.logTestingPlaygroundURL();
    // expect(
    //   screen.queryByLabelText(/Link Para meu Github/i)
    // ).toBeInTheDocument();
    // expect(
    //   screen.queryByLabelText(/Link Para meu Linkedin/i)
    // ).toBeInTheDocument();
    // expect(
    //   screen.queryByLabelText(/Realizar um contato via Email/i)
    // ).toBeInTheDocument();
  });
});
