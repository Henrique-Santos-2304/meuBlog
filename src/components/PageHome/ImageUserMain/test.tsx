import ImageUserMain from ".";
import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";
import { mockTypes } from "./mockImageUserProps";

jest.mock("@chakra-ui/react", () => {
  const originalModule = jest.requireActual("@chakra-ui/react");

  return {
    __esModule: true,
    ...originalModule,
    useMediaQuery: jest.fn().mockImplementation(() => [true]),
  };
});

describe("<ImageUserMain />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<ImageUserMain {...mockTypes} />);
    expect(container).toMatchSnapshot();
  });
  it("should have a image on the screen ", () => {
    renderTheme(<ImageUserMain {...mockTypes} />);
    expect(screen.getByRole("img")).toBeInTheDocument();
  });
});
