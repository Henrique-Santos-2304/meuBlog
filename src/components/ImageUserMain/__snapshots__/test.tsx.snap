// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageUserMain /> should render  1`] = `
.c0 svg {
  font-size: 3rem;
  border-radius: 50%;
  cursor: pointer;
}

.c0 svg::hover {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
}

.c0 svg#github {
  background: #000;
  color: #E5E7EB;
  border: 0.2rem solid #000;
}

.c0 svg#linkedin {
  color: #E5E7EB;
  font-size: 3.6rem;
}

.c0 svg#email {
  color: #E8EB33;
  font-size: 3.2rem;
}

<div>
   
  <div
    class="css-1odp1ew"
  >
    <img
      class="chakra-image css-14je8p7"
      src="https://res.cloudinary.com/defnibbpl/image/upload/v1633360607/large_henrique_techs_89a8aae1bc.jpg"
    />
    <nav
      class="c0"
    >
      <div
        class="chakra-stack css-chsok2"
      >
        <a
          aria-label="Link Para meu Github"
          class="chakra-link css-f4h6uy"
          href="https://github.com/Henrique-Santos-2304"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            fill="currentColor"
            height="1em"
            id="github"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              /Henrique-Santos-2304
            </title>
            <path
              d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
            />
          </svg>
        </a>
        <a
          aria-label="Link Para meu Linkedin"
          class="chakra-link css-f4h6uy"
          href="https://www.linkedin.com/in/henriquewebdeveloper/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            baseProfile="tiny"
            fill="currentColor"
            height="1em"
            id="linkedin"
            stroke="currentColor"
            stroke-width="0"
            version="1.2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              in/henriquewebdeveloper/
            </title>
            <g>
              <path
                d="M10.033 15.3h-1.6v-5.199h1.6v5.199zm-.8-5.866c-.577 0-.866-.267-.866-.8 0-.223.082-.412.25-.567.166-.155.371-.233.616-.233.577 0 .866.268.866.801s-.288.799-.866.799zm6.734 5.866h-1.633v-2.9c0-.755-.268-1.133-.801-1.133-.422 0-.699.211-.834.633-.043.067-.066.201-.066.4v3h-1.633v-3.533c0-.8-.012-1.355-.033-1.666h1.4l.1.699c.367-.556.9-.833 1.633-.833.557 0 1.006.194 1.35.583.346.389.518.95.518 1.684v3.066zM12 21c-4.963 0-9-4.037-9-9s4.037-9 9-9 9 4.037 9 9-4.037 9-9 9zm0-16c-3.859 0-7 3.141-7 7s3.141 7 7 7 7-3.141 7-7-3.141-7-7-7z"
              />
            </g>
          </svg>
        </a>
        <svg
          aria-label="Envio de email via site"
          fill="currentColor"
          height="1em"
          id="email"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            henrique.multitech@gmail.com
          </title>
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
          />
        </svg>
      </div>
    </nav>
  </div>
</div>
`;
